libdrizzle/pack.cc

```diff
@@ -481,3 +481,24 @@ static drizzle_return_t _pack_scramble_hash(drizzle_st *con,
 
   return DRIZZLE_RETURN_OK;
 }
+
+void drizzle_unpack_error(drizzle_st *con)
```

Comment:
Hmm, `con->buffer_size >= con->packet_size` is guaranteed so shouldn't be a problem to do 
```c++
  con->buffer_ptr+= con->packet_size;
  con->buffer_size-= con->packet_size;
  con->packet_size= 0;
```